src/scripts/farming/Plot.ts

```diff
@@ -2,43 +2,434 @@ class Plot implements Saveable {
     saveKey = '';
     defaults = {
         isUnlocked: false,
-        boosted: false,
         berry: BerryType.None,
-        timeLeft: 0,
+        age: 0,
+        mulch: MulchType.None,
+        mulchTimeLeft: 0,
     };
 
     _isUnlocked: KnockoutObservable<boolean>;
-    _boosted: KnockoutObservable<boolean>;
     _berry: KnockoutObservable<BerryType>;
-    _timeLeft: KnockoutObservable<number>;
+    _age: KnockoutObservable<number>;
+
+    _mulch: KnockoutObservable<MulchType>;
+    _mulchTimeLeft: KnockoutObservable<number>;
+
+    _auras: KnockoutObservable<number>[];
+
     formattedTimeLeft: KnockoutComputed<string>;
+    formattedMulchTimeLeft: KnockoutComputed<string>;
+    formattedAuras: KnockoutComputed<string>;
+
+    auraGrowth: KnockoutComputed<number>;
+    auraHarvest: KnockoutComputed<number>;
+    auraMutation: KnockoutComputed<number>;
+    auraReplant: KnockoutComputed<number>;
+
     isEmpty: KnockoutComputed<boolean>;
+    isMulched: KnockoutComputed<boolean>;
     stage: KnockoutComputed<number>;
-    notified: boolean;
+    tooltip: KnockoutComputed<string>;
+    notifications: FarmNotificationType[];
 
-    constructor(isUnlocked: boolean, boosted: boolean, berry: BerryType, timeLeft: number) {
+    constructor(isUnlocked: boolean, berry: BerryType, age: number, mulch: MulchType, mulchTimeLeft: number) {
         this._isUnlocked = ko.observable(isUnlocked);
-        this._boosted = ko.observable(boosted);
         this._berry = ko.observable(berry);
-        this._timeLeft = ko.observable(timeLeft);
+        this._age = ko.observable(age);
+        this._mulch = ko.observable(mulch);
+        this._mulchTimeLeft = ko.observable(mulchTimeLeft);
+
+        this._auras = [];
+        this._auras[AuraType.Growth] = ko.observable(1);
+        this._auras[AuraType.Harvest] = ko.observable(1);
+        this._auras[AuraType.Mutation] = ko.observable(1);
+        this._auras[AuraType.Replant] = ko.observable(1);
 
         this.formattedTimeLeft = ko.pureComputed(function () {
-            return GameConstants.formatTime(Math.ceil(this.timeLeft) / App.game.oakItems.calculateBonus(OakItems.OakItem.Sprayduck));
+            if (this.berry === BerryType.None) {
+                return '';
+            }
+            for (let i = 0;i < 5;i++) {
+                if (this.age < this.berryData.growthTime[i]) {
```

Comment:
I think it would be clearer to do
```js
const growthTime = this.berryData.growthTime.find(t => this.age < t)
```
and then replace `this.berryData.growthTime[i]` with `growthTime` in the rest of the calculation, to avoid returning from the middle of a for loop.